src/app/code/community/Shopgate/Framework/Model/Shopgate/Cart/Validation/Stock/Simple.php

```diff
@@ -98,6 +89,14 @@ public function validateStock(Mage_Sales_Model_Quote_Item $item, $priceInclTax,
             $stockItem->setQty(
                 (int)($item->getQtyToAdd() / $stockItem->getQtyIncrements()) * $stockItem->getQtyIncrements()
             );
+        } elseif (!$product->isAvailable()) {
```

Comment:
may need to double check, this code may exist in m2 extension as well. Although I refactored it quite a bit if I remember correctly.